// Copyright 2018 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//
// See platform2/media_perception/mojom/frame_perception.mojom for relevant comments on
// data fields.

syntax = "proto2";

package mri;

import "common.proto";

enum EntityType {
  ENTITY_TYPE_UNKNOWN = 0;
  FACE = 1;
  PERSON = 2;
  MOTION_REGION = 3;
  LABELED_REGION = 4;
}

message Entity {
  optional EntityType type = 1;
  optional string label = 2;
  optional NormalizedBoundingBox bounding_box = 3;
  optional float confidence = 4;
  optional Distance depth = 5;
}

enum FramePerceptionType {
  FRAME_PERCEPTION_TYPE_UNKNOWN = 0;
  FACE_DETECTION = 1;
  PERSON_DETECTION = 2;
  MOTION_DETECTION = 3;
}

message FramePerception {
  optional int64 frame_id = 1;
  optional int64 timestamp_us = 2;
  repeated Entity entities = 3;
  repeated FramePerceptionType perception_types = 4;
}
